### DISCUSSION TITLE
The server return TEXT/HTML instead of JSON in a NEXT.JS 14 app.

### DISCUSSION THREAD

**Discussion by YonatanAriel** [vercel/community#5248] [Help]
### Summary

I deploy a Next.js app to Vercel and the server returns text/html instead of JSON. Because of וt I can't map over the arrays in my app. Everything works fine on localhost:3000 and I get json from the server.
[package.json](https://github.com/vercel/community/files/13730553/package.json)

### Example

Link to GitHub - https://github.com/YonatanAriel/DevFolio

### Steps to Reproduce

An example of a fetch request -
export const getAllUsers = async () => {
  const URL = getUrl();
 const res = await fetch(`${URL}/api/users`, {
    next: { revalidate: 20 },
  });
  return await res.json();
}
 function getUrl() {
  return process.env.NODE_ENV === "production"
    ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`
    : process.env.NEXT_PUBLIC_BASE_URL;
}
server route example - import { NextResponse } from "next/server";
import connectToDB from "../../../DL/connectToDB";
import { User } from "../../../DL/models/User.model";

export async function GET(req) {
  await connectToDB();
  const allUsers = await User.find();
  return NextResponse.json(allUsers, { status: 200 });
}

**Comment by contigen** [comment]
Hi! Iʼm in the same boat as you, did you find a fix?

**Reply by YonatanAriel** [reply]
Couldn't find a solution... if you are on to something , please let me know.

**Comment by julioklein** [comment]
Just in case anyone else encounters this issue

In my case, doing a test with insomnia, I found that Vercel returned a page for authentication

I needed to go to project settings in vercel -> Deployment Protection -> Vercel Authentication -> disable

After saving and testing again with insomnia, it worked

**Reply by YonatanAriel** [reply]
thanks, it didn't help me unfortunately.

**Reply by edfrommathclass** [reply]
@YonatanAriel did you ever find a fix for this? I'm facing the same issue. Was getting a failed to fetch error until I disable the deployment protection but now the API is returning a text/html value.
